src/axom/core/examples/core_array_perf.cpp

```diff
@@ -0,0 +1,975 @@
+
+#include "axom/fmt/format.h"
+#include "axom/CLI11.hpp"
+
+// C/C++ includes
+#include <iostream>
+#include <vector>
+
+///////////////////////////////////////////////////////////////
+/// Struct to parse and store the input parameters
+struct InputParams
+{
+public:
+  using RuntimePolicy = axom::runtime_policy::Policy;
+
+  // Array shape.
+  std::vector<axom::IndexType> shape;
+  axom::IndexType ghostWidth = 1;
+  std::vector<axom::IndexType> paddedShape;
+  std::vector<axom::IndexType> idxBegin;
+  std::vector<axom::IndexType> idxEnd;
+  axom::IndexType realSize;
+  axom::IndexType paddedSize;
+
+  // Array stride order (same length as shape)
+  std::vector<std::uint32_t> dataSlowestDirections;
+  axom::ArrayStrideOrder dataOrder = axom::ArrayStrideOrder::ARBITRARY;
+
+  RuntimePolicy runtimePolicy = RuntimePolicy::seq;
+
+  axom::IndexType repCount = 10;
+
+private:
+  bool _verboseOutput {false};
+  const std::map<std::string, int> strideValidator {
+    {"row", int(axom::ArrayStrideOrder::ROW)},
+    {"col", int(axom::ArrayStrideOrder::COLUMN)}};
+
+public:
+  bool isVerbose() const { return _verboseOutput; }
+
+  void parse(int argc, char** argv, axom::CLI::App& app)
+  {
+    app.add_option("-p, --policy", runtimePolicy)
+      ->description("Set runtime policy for test")
+      ->capture_default_str()
+      ->transform(
+        axom::CLI::CheckedTransformer(axom::runtime_policy::s_nameToPolicy));
+
+    app.add_flag("-v,--verbose,!--no-verbose", _verboseOutput)
+      ->description("Enable/disable verbose output")
+      ->capture_default_str();
+
+    app.add_option("-s, --shape", shape)->description("Array shape")->expected(1, 4);
+
+    app.add_option("-g, --ghost", ghostWidth)->description("Ghost width");
+
+    app.add_option("-r, --repCount", repCount)
+      ->description("Number of repetitions to run");
+
+    auto* dataOrderOption =
+      app.add_option("--dataOrder", dataOrder)
+        ->description("Stride order of array data.")
+        ->transform(axom::CLI::CheckedTransformer(strideValidator))
+        ->expected(1, 4);
+
+    app.add_option("--dataSlowestDirections", dataSlowestDirections)
+      ->description(
+        "Array data stride directions, from slowest to fastest."
+        "  Must be same length as shape.")
+      ->excludes(dataOrderOption);
+
+    app.get_formatter()->column_width(60);
+
+    app.parse(argc, argv);
+
+    if(shape.empty())
+    {
+      std::cerr << "You must specify shape (1-4 integers)." << std::endl;
+      std::abort();
+    }
+
+    /*
+      If dataSlowestDirections is specified, it must match length of shape.
+      If neither is specified, default to row-major ordering.
+    */
+    if(!dataSlowestDirections.empty())
+    {
+      if(dataSlowestDirections.size() != shape.size())
+      {
+        std::cerr << "slowestDimension size (" << dataSlowestDirections.size()
+                  << ") must match shape size (" << shape.size() << ")."
+                  << std::endl;
```

Comment:
Minor: Please consider increasing the slic and fmt usage in this example.

Reply:
I'll use `fmt` more but I don't think I can use `slic` here because `core` can't depend on `slic`.

Reply:
Good call.  This isn't required for the PR, so don't feel obligated to make the change.

`core` can't depend on `slic`, but I don't see any reason that `core`'s examples can't depend on `slic`
(obviously, the example should check that `slic` is enabled)